### ISSUE TITLE
CSI plugin fails to authenticate after upgrade of kubernetes from 13 (csi 0.4.0) to 14 (csi 1.1.0) 

### ISSUE THREAD

**Issue by krasaee** [digitalocean/csi-digitalocean#155]
### What did you do? (required. The issue will be **closed** when not provided.)

1. upgraded kubernetes nodes from version 13 to 14 using kubeadm
2. upgraded csi plugin from 0.4.0 by simply applying v1.1.0 via kubectl
3. everything appeared to work for a while but now receiving (401 Unable to authenticate you)
4. manually removed traces of do-csi, daemonsets, roles and so forth. 
5. deleted secret and applied new secret, checked using doctl that it is correct after running "export DIGITALOCEAN_ACCESS_TOKEN=$(kubectl get secrets -n kube-system digitalocean -o json | jq -r '.data["access-token"]'| base64 --decode)"
and then doctl compute volume list
7. created simple pvc but receiving 401 unauthorized
8. triple checked to do secret

### What did you expect to happen?

1. expect pvc to provision as it was working with 0.4.0 of csi plugin on kube 13

### Configuration (**MUST** fill this out):

* system logs:

`kubectl logs csi-do-controller-0 -n kube-system csi-provisioner`

```
I0505 10:16:40.392423       1 controller.go:1196] provision "default/csi-pvc" class "do-block-storage": started
I0505 10:16:40.395540       1 controller.go:442] CreateVolumeRequest {Name:pvc-baef1cc8-6f1e-11e9-9f09-1e6b332d2b41 CapacityRange:required_bytes:5368709120  VolumeCapabilities:[mount:<fs_type:"ext4" > access_mode:<mode:SINGLE_NODE_WRITER > ] Parameters:map[] Secrets:map[] VolumeContentSource:<nil> AccessibilityRequirements:<nil> XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}
I0505 10:16:40.395622       1 connection.go:180] GRPC call: /csi.v1.Controller/CreateVolume
I0505 10:16:40.397285       1 event.go:209] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"csi-pvc", UID:"baef1cc8-6f1e-11e9-9f09-1e6b332d2b41", APIVersion:"v1", ResourceVersion:"37126277", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "default/csi-pvc"
I0505 10:16:40.395631       1 connection.go:181] GRPC request: {"capacity_range":{"required_bytes":5368709120},"name":"pvc-baef1cc8-6f1e-11e9-9f09-1e6b332d2b41","volume_capabilities":[{"AccessType":{"Mount":{"fs_type":"ext4"}},"access_mode":{"mode":1}}]}
I0505 10:16:40.504103       1 connection.go:183] GRPC response: {}
I0505 10:16:40.504787       1 connection.go:184] GRPC error: rpc error: code = Internal desc = GET https://api.digitalocean.com/v2/volumes?name=pvc-baef1cc8-6f1e-11e9-9f09-1e6b332d2b41&region=fra1: 401 Unable to authenticate you
I0505 10:16:40.504866       1 controller.go:979] Final error received, removing PVC baef1cc8-6f1e-11e9-9f09-1e6b332d2b41 from claims in progress
W0505 10:16:40.504889       1 controller.go:886] Retrying syncing claim "baef1cc8-6f1e-11e9-9f09-1e6b332d2b41", failure 6
E0505 10:16:40.504923       1 controller.go:908] error syncing claim "baef1cc8-6f1e-11e9-9f09-1e6b332d2b41": failed to provision volume with StorageClass "do-block-storage": rpc error: code = Internal desc = GET https://api.digitalocean.com/v2/volumes?name=pvc-baef1cc8-6f1e-11e9-9f09-1e6b332d2b41&region=fra1: 401 Unable to authenticate you
I0505 10:16:40.505409       1 event.go:209] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"csi-pvc", UID:"baef1cc8-6f1e-11e9-9f09-1e6b332d2b41", APIVersion:"v1", ResourceVersion:"37126277", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' failed to provision volume with StorageClass "do-block-storage": rpc error: code = Internal desc = GET https://api.digitalocean.com/v2/volumes?name=pvc-baef1cc8-6f1e-11e9-9f09-1e6b332d2b41&region=fra1: 401 Unable to authenticate you
```

`kubectl describe pvc csi-pvc`
```
Events:
  Type       Reason                Age                     From                                                                                Message
  ----       ------                ----                    ----                                                                                -------
  Normal     Provisioning          4m8s (x9 over 8m22s)    dobs.csi.digitalocean.com_csi-do-controller-0_7ba7ca34-6f17-11e9-8b51-9675be27db13  External provisioner is provisioning volume for claim "default/csi-pvc"
  Warning    ProvisioningFailed    4m8s (x9 over 8m22s)    dobs.csi.digitalocean.com_csi-do-controller-0_7ba7ca34-6f17-11e9-8b51-9675be27db13  failed to provision volume with StorageClass "do-block-storage": rpc error: code = Internal desc = GET https://api.digitalocean.com/v2/volumes?name=pvc-baef1cc8-6f1e-11e9-9f09-1e6b332d2b41&region=fra1: 401 Unable to authenticate you
  Normal     ExternalProvisioning  2m46s (x26 over 8m22s)  persistentvolume-controller                                                         waiting for a volume to be created, either by external provisioner "dobs.csi.digitalocean.com" or manually created by system administrator
Mounted By:  <none>
```

Please provide the following logs:

```

kubectl cluster-info dump > kubernetes-dump.log
```

This will output everthing from your cluster.  Please use a private gist via
https://gist.github.com/ to share this dump with us

* manifests, such as pvc, deployments, etc.. you used to reproduce:

```apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: csi-pvc
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
  storageClassName: do-block-storage```

Please provide the **total** set of manifests that are needed to reproduce the
issue. Just providing the `pvc` is not helpful. If you cannot provide it due
privacy concerns, please try creating a reproducible case.


* CSI Version:
v1.1.0

* Kubernetes Version:

v1.14.1

* Cloud provider/framework version, if applicable (such as Rancher):

**Comment by krasaee** [comment]
Updated
I was able to get pvc functionality working again, but I had to forcefully/manually delete all the csi plugins from all nodes via /var/lib/kubelet/plugins and ./plugins_registry, including removal of referenced values in kubelet get csidrivers, roles, clusterroles, bindings, lingering service accounts from old flex drivers, etcetera. 

After removal, I applied csi v0.4.0 and the pvc functionality returned to normal. 

I presume there is a version conflict between old and new, maybe related to lingering old binaries? I will likely reattempt to delete everything again and apply v1.1.0 on kubernetes v1.14.1
